graph/graph.go

```diff
@@ -82,6 +82,9 @@ type Graph struct {
 	imageMutex       imageMutex // protect images in driver.
 	retained         *retainedLayers
 	tarSplitDisabled bool
+
+	parentRefs      map[string]uint16
```

Comment:
So, if we want to be superoptimized, then let's just use `uint32` and `atomic.AddUint32` without mutex